### PR TITLE
Closes #17522: Add MOZILLA_ONLINE_ADDON_EXCLUSIONS for Mozilla Online builds

### LINKED ISSUE #17522: [Feature China]Requirement of add-on list in China
There's an issue before about this topic #16093
However, we've found that creating a different add-on collection is not the best choice, since we hope that users are able to install and use all the add-ons even if some of them are not in our recommendation list. What we truly care about is we **cannot recommend** some of the add-ons due to some legal reason and we don't want to interfere with users more than that.
So our goal is not creating  a totally different collection with global builds, but just using the same collection and not recommending some of them.
It may involve some modification of android-components. I have modified a-c locally to build a fenix reaching this goal, this can be a start of our discussion: [Filter recommendation add-on list for Mozilla Online builds](https://github.com/MozillaOnline/android-components/commit/d22bd48379155e0db4db44bd036812d79b17b1ea)

### PR DESCRIPTION
**Pull request by rxumoz**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by Amejia481 — APPROVED** [review]
LGTM!

File: app/build.gradle [hunk 1]
```diff
@@ -41,6 +41,18 @@ android {
...
+                "{" +
+                        "\"uBlock0@raymondhill.net\"," +
+                        "\"firefox@ghostery.com\"," +
+                        "\"jid1-MnnxcxisBPnSXQ@jetpack\"," +
+                        "\"adguardadblocker@adguard.com\"," +
+                        "\"foxyproxy@eric.h.jung\"," +
+                        "\"{73a6fe31-595d-460b-a920-fcc0f8843232}\"," +
+                        "\"jid1-BoFifL9Vbdl2zQ@jetpack\"," +
+                        "\"woop-NoopscooPsnSXQ@jetpack\"" +
+                        "}"
```
Comment (csadilek) [inline app/build.gradle:55]: nit: indentation of the "}". Would align with opening "{".

File: app/build.gradle [hunk 2]
```diff
@@ -41,6 +41,18 @@ android {
         buildConfigField "String", "AMO_BASE_URL", "\"https://addons.mozilla.org\""
         buildConfigField "String", "AMO_COLLECTION_NAME", "\"7dfae8669acc4312a65e8ba5553036\""
         buildConfigField "String", "AMO_COLLECTION_USER", "\"mozilla\""
+        // This should be exluded add-on ID array for Mozilla Online
```
Comment (csadilek) [inline app/build.gradle:44]: nit: maybe just: These add-ons should be excluded for Mozilla Online builds.

File: app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt [hunk 3]
```diff
@@ -105,14 +107,20 @@ class AddonsManagementFragment : Fragment(R.layout.fragment_add_ons_management)
         lifecycleScope.launch(IO) {
             try {
                 val addons = requireContext().components.addonManager.getAddons(allowCache = allowCache)
+                var excludedAddonIDs = emptyList<String>()
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt:110]: nit: We could use a val in a single statement, see below.

File: app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt [hunk 4]
```diff
@@ -105,14 +107,20 @@ class AddonsManagementFragment : Fragment(R.layout.fragment_add_ons_management)
...
                 lifecycleScope.launch(Dispatchers.Main) {
                     runIfFragmentIsAttached {
                         if (!shouldRefresh) {
+                            if (Config.channel.isMozillaOnline &&
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt:114]: We could also just write:
```kotlin
val excludedAddonIds = if (Config.channel.isMozillaOnline && 
  BuildConfig.MOZILLA_ONLINE_ADDON_EXCLUSIONS.isNullOrEmpty()) {
    BuildConfig.MOZILLA_ONLINE_ADDON_EXCLUSIONS.toList()
} else {
  emptyList<String>()
}
```

**Review by csadilek — APPROVED** [review]
👍 Just some nits to consider.

**Review by csadilek — APPROVED** [review]

**Pull request merged**